File: app/src/main/java/com/example/administrator/p2pdemotext/Base/HomePresenter.java
package com.example.administrator.p2pdemotext.Base;

import com.example.administrator.p2pdemotext.Base.Bean;
import com.example.base.BasePresenter;
import com.example.base.IBasePresenter;
import com.example.net.ResEntity;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.HashMap;
import java.util.List;

public class HomePresenter extends BasePresenter<Bean> {
    @Override
    public String getpath() {
        return "dish_list.php";
    };

    @Override
    public HashMap<String,String> getParmas(){
        HashMap<String,String> paramMap=new HashMap<>();
        paramMap.put("stage_id", "1");
        paramMap.put("limit", "20");
        paramMap.put("page", "1");
        return paramMap;
    }
    @Override
    public Type getBeanType() {
        return new TypeToken<ResEntity<List<Bean>>>(){}.getType();
    }

    @Override
    public void attachView(IBasePresenter ibaseView) {

    }
    @Override
    public boolean isList(){
        return false;
    }
}

File: app/src/main/java/com/example/p2pfinancial/HomePresenter.kt
package com.example.p2pfinancial

import com.example.base.BasePresenter
import com.example.net.Constant
import com.example.net.ResEntity
import com.google.gson.reflect.TypeToken
import java.lang.reflect.Type
import java.util.HashMap

class HomePresenter : BasePresenter<HomeBean>() {

    override fun getPath(): String {
        return "dish_list.php"
    }

    override fun getType(): Type {
        return object : TypeToken<ResEntity<List<HomeBean>>>() {}.type
    }

    override fun isList(): Boolean {
        return true
    }

    override fun getQueryMap(): HashMap<String, String> {
        val map = hashMapOf<String, String>()
        map.put("stage_id", "1")
        map.put("limit", "20")
        map.put("page", "1")
        return map
    }
}